causal_testing/main.py

```diff
@@ -281,17 +281,26 @@ def create_causal_test(self, test: dict, base_test: BaseTestCase) -> CausalTestC
         if estimator_class is None:
             raise ValueError(f"Unknown estimator: {test['estimator']}")
 
+        # Filter the data if a test-specific query exists, otherwise use the original data
+        test_query = test.get("query", self.query)
```

Comment:
Please could this be conjoined rather than overriding? I can imagine a use case where someone has a general query (i.e. a modelling scenario) and wants a test specific query too (i.e. accessing a particular value in the CATE). Such a causal question might be "How does closing schools affect the number of old people with COVID for modelling scenarios in the UK?", where "modelling scenarios in the UK" is the general query applied to every test case and "old people" is the test specific query. You might then also have causal tests for children, young adults, and middle aged folks. I actually did this in my CARLA case study, except I didn't need a modelling scenario query here since I could control my dataset to only contain relevant data.